apps/condo/domains/notification/access/PushToken.js

```diff
@@ -0,0 +1,55 @@
+// @ts-nocheck
+/**
+ * Generated by `createschema notification.PushToken PushToken`
+ */
+
+const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')
+
+async function canReadPushTokens ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin) return {}
+
+    return {
+        context: {
+            OR: [
+                {
+                    owner: { id: user.id },
+                },
+                // may be extended in future
+            ],
+        },
+    }
+}
+
+async function canManagePushTokens ({ authentication: { item: owner }, originalInput, operation }) {
+    if (owner && owner.deletedAt) return false
+    if (owner && owner.isAdmin) return true
+    if (operation === 'create') {
+        // NOTE: everyone can create push tokens, including anonymous users
+        return true
+    } else if (operation === 'update') {
+        // NOTE: everyone can update every push token if they know deviceId (almost impossible to guess)
+        return {
+            context: {
+                OR: [
+                    {
+                        deviceId: { id: originalInput.deviceId },
+                    },
+                    // may be extended in future
```

Comment:
Everything can be extended in the future)
```suggestion

```

Reply:
fixed